File: src/Tiantong.Iot.Plc/Entities/PlcState.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Tiantong.Iot.Plc
{
  [Table("plc_states")]
  public class PlcState
  {
    public virtual int Id { get; set; }

    public virtual int PlcId { get; set; }

    public virtual string Name { get; set; }

    public virtual string Type { get; set; }

    public virtual string Length { get; set; }

    public virtual string Address { get; set; }

    public virtual bool IsHeartbeat { get; set; }

    public virtual int HeartbeatInterval { get; set; } = 1000;

    public virtual int HeartbeatMaxValue { get; set; } = 10000;

    public virtual bool IsCollect { get; set; }

    public virtual int CollectInterval { get; set; }

    public virtual bool OpenLog { get; set; }
  }
}

File: src/Tiantong.Iot/Tiantong.Iot.Client/src/entities/PlcState.ts
export class PlcState {
  id: number = 0
  plc_id: number = 0
  name: string = ''
  number: string = ''
  type: string = PlcStateType.uint16
  address: string = ''
  length: number = 8
  is_heartbeat: boolean = false
  heartbeat_interval: number = 1000
  heartbeat_max_value: number = 1000
  is_collect: boolean = false
  collect_interval: number = 1000
  is_read_log_on: boolean = false
  is_write_log_on: boolean = false
}

export enum PlcStateType {
  bool = 'bool',
  uint16 = 'uint16',
  int32 = 'int32',
  string = 'string'
}
